common/djangoapps/student/views.py

```diff
@@ -1827,16 +1829,22 @@ def auto_auth(request):
     create_comments_service_user(user)
 
     # Provide the user with a valid CSRF token
-    # then return a 200 response
+    # then return a 200 response unless redirect is true
     success_msg = u"{} user {} ({}) with password {} and user_id {}".format(
         u"Logged in" if login_when_done else "Created",
         username, email, password, user.id
     )
     response = HttpResponse(success_msg)
     response.set_cookie('csrftoken', csrf(request)['csrf_token'])
+    if redirect_when_done:
+        # Redirect to course info page if course_id is known
+        if course_id:
+            return redirect(reverse('info', kwargs={'course_id': unicode(course_id)} ))
+        # Otherwise redirect to dashboard
```

Comment:
Nit: It's subjective - but I'd say this comment is unnecessary due to the directness of the code below.